src/serializer/functions.js

```diff
@@ -373,7 +372,9 @@ export class Functions {
         }
       }
     }
-    return call.bind(this, globalThis, args);
+
+    let thisArg = createAbstractArgument(this.realm, "this", funcValue.expressionLocation, ObjectValue);
```

Comment:
Is it safe to assume that this has been bound to an object?

Reply:
What do you mean by that?

Reply:
It appears from experiments that the this value of a function is always an object, but the specification does not obviously require a boxed object. I guess it is safe enough, but it is hard to be quite sure.

Reply:
It is perhaps worthwhile to add a test case that supplies a non object to apply.

Reply:
The type domain I supply here is `ObjectValue`, so we are getting an `AbstractObjectValue`. That should be good enough.